NorthstarDedicatedTest/plugin_abi.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include <string>
+
+#define ABI_VERSION 1
+//clang-format off
+// I hate clang-format
+/// <summary>
+/// This enum is used for referencing the different types of objects we can pass to a plugin
+/// Anything exposed to a plugin must not a be C++ type, as they could break when compiling with a different compiler.
+/// Any ABI incompatible change must increment the version number.
+/// Nothing must be removed from this enum, only appended. When it absolutely necessary to deprecate an object, it should return UNSUPPORTED
+/// when retrieved
+/// </summary>
+enum PluginObject
+{
+	UNSUPPORTED = 0,
+	GAMESTATE = 1,
+	SERVERINFO = 2,
+	PLAYERINFO = 3,
+	DUMMY = 0xFFFF
+};
+
+enum GameStateInfoType
+{
+	ourScore,
+	secondHighestScore,
+	highestScore,
+	connected,
+	loading,
+	map,
+	mapDisplayName,
+	playlist,
+	playlistDisplayName,
+	players
```

Comment:
Add numbers here too so it's clear it can't be changed. Also add the dummy so we can safely extend this later.